src/utils/PrivateRoute.tsx

```diff
@@ -1,13 +1,32 @@
-import React, { ReactNode, useContext } from 'react';
+import React, { ReactNode, useContext, useState } from 'react';
 import { Navigate, useLocation } from 'react-router-dom';
 import { UserContext } from '../hook/useAuth';
+import checkTokenExpiration from '../utils/tokenValidation';
 
 interface Props {
   children: ReactNode;
   // any props that come into the component
 }
 
 function CheckRole({ children, ...props }: Props) {
+  const [token, setToken] = useState(true);
+  const { logout } = useContext(UserContext);
+
+
+   const interval  = setInterval(() =>{ 
+    checkTokenExpiration();        
+        const promise1 = Promise.resolve(checkTokenExpiration());
+        promise1.then((value) => {
+          setToken(value)
+        });
+          // eslint-disable-next-line no-console
+          console.log(token)
+        if(!token){
+          logout();
+          clearInterval(interval);
+        }
+  },2000)
+
```

Comment:
This is not the best way to implement this feature. You do not need to keep checking if the token has expired so you can logout a user, instead, you wanna wait for them to perform a certain action then when they send a request along its token, check if it has expired and re-route them to login page *with a message that they got logged out due to inactivity*